UI/Controllers/MentorController.cs

```diff
@@ -17,52 +17,55 @@ public class MentorController : ControllerBase
     {
         private readonly ICandidateService _candidateService;
         private readonly IEmployeeService _employeeService;
-        private readonly IInternshipTeamService _internshipTeam;
+        private readonly IInternshipTeamService _internshipTeamService;
+        private readonly ISpecializationService _specializationService;
+        private readonly IEnglishLevelService _englishLevelService;
+        private readonly ICityService _cityService;
+        private readonly IMapper _mapper;
         private readonly UserManager<User> _userManager;
 
-        public MentorController(ICandidateService candidateService, IInternshipTeamService internshipTeam, IEmployeeService employeeService, UserManager<User> userManager)
+        public MentorController(ICandidateService candidateService,
+                                ISpecializationService specializationService,
+                                IInternshipTeamService internshipTeamService,
+                                IEmployeeService employeeService,
+                                IEnglishLevelService englishLevelService,
+                                ICityService cityService,
+                                IMapper mapper,
+                                UserManager<User> userManager)
         {
             _candidateService = candidateService;
             _userManager = userManager;
             _employeeService = employeeService;
-            _internshipTeam = internshipTeam;
+            _internshipTeamService = internshipTeamService;
+            _specializationService = specializationService;
+            _englishLevelService = englishLevelService;
+            _cityService = cityService;
+            _mapper = mapper;
         }
 
-        // GET: api/<MentorController>
-        [HttpGet]
+        [HttpGet("candidates")]
         [Authorize(Roles = "admin, mentor")]
         public async Task<List<CandidateDTO>> GetCandidates()
```

Comment:
all things related to the candidates should be placed in appropariate locations, like CandidateController
UI will be confused with your endpoints:
api/mentor/getCandidates
api/candidate/getCandidates